cirq/sim/density_matrix_simulator.py

```diff
@@ -285,7 +279,7 @@ def keep(potential_op: ops.Operation) -> bool:
                             indices,
                             qid_shape=qid_shape,
                             out=state.tensor,
-                            seed=self.prng)
+                            seed=self._seed)
```

Comment:
If `self._seed` is an int, won't this line cause the same RandomState to be initialized each time it is executed? That's why I believe the correct behavior is to initialize a RandomState during initialization of the simulator and pass that object around; we should never have more than one RandomState be initialized during the course of a single simulation.